File: core/schemas.py
from pydantic import BaseModel

# DATA BASE

class SiteDataBase(BaseModel):
    headertext: str
    bodytext: str
    imageurl: str
    siteurl: str

class SiteDataCreate(SiteDataBase):
    pass

class SiteDataResponse(SiteDataBase):
    id: int

    class Config:
        from_attributes = True

# INFO PAGE DATA BASE

class InfoDataBase(BaseModel):
    headertext: str
    bodytext: str

class InfoDataResponse(InfoDataBase):
    id: int

    class Config:
        from_attributes = True

# USER BASE

class UserCreate(BaseModel):
    username: str
    password: str

class UserResponse(BaseModel):
    id: int
    username: str

    class Config:
        from_attributes = True

class Token(BaseModel):
    access_token: str
    token_type: str

File: core/schemas.go
package core

type SiteDataCreate struct {
	HeaderText string `json:"headertext" binding:"required"`
	BodyText   string `json:"bodytext" binding:"required"`
	ImageURL   string `json:"imageurl" binding:"required"`
	SiteURL    string `json:"siteurl" binding:"required"`
}

type LoginSchema struct {
	Username string `json:"username" binding:"required"`
	Password string `json:"password" binding:"required"`
}
